cmake/MiscCheck.cmake

```diff
@@ -142,66 +142,7 @@ endif()
 # a third party library (like Protobuf), mention it in the comment as
 # "THIRD_PARTY_NAME related"
 # From https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/
-if(${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC")
-  add_compile_options(
-      ##########################################
-      # Protobuf related. Cannot remove.
-      # This is directly copied from
-      #     https://github.com/google/protobuf/blob/master/cmake/README.md
-      ##########################################
-      /wd4018 # 'expression' : signed/unsigned mismatch
```

Comment:
This code is ancient and probably should go, but have you checked if it actually suppresses some errors/warnings? Can you compare size of build log before/after the change?

Reply:
I find MSVC flags set in multiple locations. Some are redundant. I will recheck them when the tests finish.